typescript/lib/bulk_command_execution/v0_bulk_command_api/v0_bulk_commands.ts

```diff
@@ -0,0 +1,249 @@
+    	
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitRepartition(): Result<null, Error> {
+    	let args: RepartitionArgs;
+    	
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitExecCommand(): Result<null, Error> {
+        let args: ExecCommandArgs;
+     
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitWaitForEndpointAvailability(): Result<null, Error> {
+    	let args: WaitForEndpointAvailabilityArgs;
+    	
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public visitExecuteBulkCommands(): Result<null, Error> {
+        let args: ExecuteBulkCommandsArgs;
+
+        try {
+            args = JSON.parse(this.bytesToDeserialize);
+        } catch(jsonErr) {
+            return err(jsonErr);
+        }
+
+        this.deserializedCommandArgsPtr = args;
+        return ok(null);
+    }
+
+    public getDeserializedCommandArgs(): protobuf.Message {
+        return this.deserializedCommandArgsPtr;
+    }
+}
+
+// ====================================================================================================
+//                                        Serializable Command
+// ====================================================================================================
+
+//TODO - Potentially REMOVE
+// class InterstitialStruct {
+//     private readonly type: V0CommandType;
+//     private argsBytes: string; //TODO (comment) - original type was json.RawMessage ; couldn't find equivalent in typescript
+
+//     constructor(){}
+
+//     //TODO (comment)- getter and setter instead of giving direct access is alright change from golang, is this okay?
+//     public getType(): V0CommandType {
+//         return this.type;
+//     }
+    
+//     public getArgsBytes(): string {
+//         return this.argsBytes;
+//     }
+
+//     public setArgsBytes(newArgsBytes: string): void {
+//         this.argsBytes = newArgsBytes;
+//     }
+// }
+
+// Used for serializing
+export class V0SerializableCommand {
+    private type: V0CommandType;
+    
+    // The only allowed objects here are from the bindings generated from the .proto file
+    private argsPtr: protobuf.Message;
+    
+    //TODO (comment) - added getter and setters instead of giving direct access, is this okay?
```

Comment:
I'd prefer if we do the setting via constructor (allows you to make things `readonly`, i.e. makes it immutable, which is always better than mutable)